rfcs/experimental/dynamic_selection/README.md

```diff
@@ -0,0 +1,516 @@
+| ------- | ----------- |
+| `policy_traits<T>::selection_type`, `selection_t<T>` | The wrapped select type returned by `T`. Must satisfy [Selection](#selection_req_id). |
+| `policy_traits<T>::resource_type`, `resource_t<T>` | The backend defined resource type that is passed to the user function object. Calling `unwrap` an object of type `selection_t<T>` returns an object of type `resource_t<T>`. |
+| `policy_traits<T>::wait_type`, `wait_type_t<T>` | The backend type that is returned by the user function object. Calling `unwrap` on an object that satisfies [Submission](#submission_req_id) returns on object of type `wait_type_t<T>`. |
+
+*Policy traits are defined in `include/oneapi/dpl/internal/dynamic_selection_impl/policy_traits.h`.
+
+The current implementation of these traits depends on types defined in the Policy:
+
+```cpp
+  template <typename Policy>
+  struct policy_traits
+  {
+      using selection_type = typename std::decay_t<Policy>::selection_type;
+      using resource_type = typename std::decay_t<Policy>::resource_type;
+      using wait_type = typename std::decay_t<Policy>::wait_type;
+  };
+```
+| *Optional* | Description |
+| --------------------- | ----------- |
+| `p.submit_and_wait(s, f, args…)` | Returns `void`. The function invokes `f` and waits for the `wait_t<T>` it returns to complete. |
+| `p.submit(f, args…)` | Returns `submission_t<T>` that satisfies [Submission](#submission_req_id). The function invokes `f` but does not wait for the `wait_t<T>` it returns to complete. |
+| `p.submit_and_wait(f, args…)` | Returns `void`. The function invokes `f` and waits for the `wait_t<T>` it returns to complete. |
+
+<a id="selection_req_id"></a>
+### Selection
+
+The type `T` satisfies *Selection* for a given [Policy](#policy_req_id) `p` if given,
+
+- `s` an arbitrary identifier of type `T`
+- `i` an object of type `Info` where `execution_info_v<Info>` is `true`
+- 'v' an object of type `V` and `is_same_v<Info::value_type, V>` is `true`
+
+| *Must* be well-formed | Description |
+| --------------------- | ----------- |
+| `s.unwrap()` | Returns `resource_t<T>` that should represent one of the resources returned by `p.get_resources()` for the Policy `p` that generated `s`. |
+| `s.policy()` | Returns the Policy `p` that was used to make the selection. |
+| `s.report(i)` | Returns `void`. Notifies policy that an execution info event has occurred. |
+| `report_execution_info<T, Info>::value`, `report_execution_info_v<T,Info>` | `true` if this selection needs the backend to report the Info. `false` otherwise. |
+
+<a id="submission_req_id"></a>
+### Submission
+
+The type `T` satisfies *Submission* for a given [Policy](#policy_req_id) `p` if given,
+
+- `s` an arbitrary identifier of type `T`
+
+| *Must* be well-formed | Description |
+| --------------------- | ----------- |
+| `s.get_policy()` | Returns an object that satisfies [Policy](#policy_req_id) and corresponds to the Policy that made the selection. |
```

Comment:
Why the method is called `get_policy()` here but a similar method of `Selection` is called `policy()`?

Also, should the description say "...that made the **submission**"?

Reply:
These names were taken from an internal design doc that was out of date. The current design has only `get_policy` and only in Selection. I will update.